pkg/cluster/gatherlogs.go

```diff
@@ -9,29 +9,50 @@ import (
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	"github.com/Azure/ARO-RP/pkg/cluster/failurediagnostics"
 	"github.com/Azure/ARO-RP/pkg/util/steps"
 )
 
+type diagnosticStep struct {
+	f      func(context.Context) (interface{}, error)
+	isJSON bool
+}
+
 func (m *manager) gatherFailureLogs(ctx context.Context) {
-	for _, f := range []func(context.Context) (interface{}, error){
-		m.logClusterVersion,
-		m.logNodes,
-		m.logClusterOperators,
-		m.logIngressControllers,
+	d := failurediagnostics.NewFailureDiagnostics(m.log, m.env, m.doc, m.virtualMachines, m.storage)
+
+	for _, f := range []diagnosticStep{
+		{f: m.logClusterVersion, isJSON: true},
+		{f: m.logNodes, isJSON: true},
+		{f: m.logClusterOperators, isJSON: true},
+		{f: m.logIngressControllers, isJSON: true},
+		{f: d.LogAzureInformation, isJSON: false},
```

Comment:
This feedback doesn't make a lot of sense to me. What's the need for adding another function?

Reply:
I mean because we added `d.LogAzureInformation` in `gatherFailureLogs`, we need `isJSON` field and some conditional branches in here.

If we make a new method instead of `gatherFailureLogs`, and call `d.LogAzureInformation` there not in `gatherFailureLogs`, we can keep this method simple.

Reply:
I think that the existing process of dumping out YAML/JSON is a bit unfriendly for SREs, especially when it comes to the ClusterOperators output. I think it would be better to move some of these to more formatted output, as well as have the ability to automatically analyse the logs (e.g. the serial console logs) and output formatted results there. I'm not sure if wrapping this func in another will make it overall more simple, but I do think another pass at this whole area would be a welcome change in a separate PR.